official/vision/beta/projects/panoptic_maskrcnn/tasks/panoptic_maskrcnn.py

```diff
@@ -0,0 +1,381 @@
+        segmentation_resize_eval_groundtruth=params.parser
+        .segmentation_resize_eval_groundtruth,
+        segmentation_groundtruth_padded_size=params.parser
+        .segmentation_groundtruth_padded_size,
+        segmentation_ignore_label=params.parser.segmentation_ignore_label)
+
+    reader = input_reader_factory.input_reader_generator(
+        params,
+        dataset_fn=dataset_fn.pick_dataset_fn(params.file_type),
+        decoder_fn=decoder.decode,
+        parser_fn=parser.parse_fn(params.is_training))
+    dataset = reader.read(input_context=input_context)
+
+    return dataset
+
+  def build_losses(self,
+                   outputs: Mapping[str, Any],
+                   labels: Mapping[str, Any],
+                   aux_losses: Optional[Any] = None):
+    """Build Panoptic Mask R-CNN losses."""
+    params = self.task_config.losses
+    segmentation_loss_params = params.segmentation_loss
+    use_groundtruth_dimension = \
+        segmentation_loss_params.use_groundtruth_dimension
+
+    segmentation_loss_fn = segmentation_losses.SegmentationLoss(
+        segmentation_loss_params.label_smoothing,
+        segmentation_loss_params.class_weights,
+        segmentation_loss_params.ignore_label,
+        use_groundtruth_dimension=use_groundtruth_dimension,
+        top_k_percent_pixels=segmentation_loss_params.top_k_percent_pixels)
+
+    losses = super(PanopticMaskRCNNTask, self).build_losses(
+        outputs=outputs,
+        labels=labels,
+        aux_losses=None)
+    maskrcnn_loss = losses['model_loss']
+    segmentation_loss = segmentation_loss_fn(
+        outputs['segmentation_outputs'],
+        labels['gt_segmentation_mask'])
+
+    model_loss = (
+        maskrcnn_loss + params.segmentation_weight * segmentation_loss)
+
+    total_loss = model_loss
+    if aux_losses:
+      reg_loss = tf.reduce_sum(aux_losses)
+      total_loss = model_loss + reg_loss
+
+    losses.update({
+        'total_loss': total_loss,
+        'maskrcnn_loss': maskrcnn_loss,
+        'segmentation_loss': segmentation_loss,
+        'model_loss': model_loss,
+    })
+    return losses
+
+  def build_metrics(self, training: bool = True):
+    """Build detection metrics."""
+    metrics = []
+    if training:
+      metric_names = [
+          'total_loss',
+          'rpn_score_loss',
+          'rpn_box_loss',
+          'frcnn_cls_loss',
+          'frcnn_box_loss',
+          'mask_loss',
+          'maskrcnn_loss',
+          'segmentation_loss',
+          'model_loss'
+      ]
+      for name in metric_names:
+        metrics.append(tf.keras.metrics.Mean(name, dtype=tf.float32))
+      
+      if self.task_config.segmentation_evaluation.report_train_mean_iou:
+        self.segmentation_train_mean_iou = segmentation_metrics.MeanIoU(
+            name='train_mean_iou',
+            num_classes=self.task_config.model.num_classes,
+            rescale_predictions=False,
+            dtype=tf.float32)
+
+    else:
```

Comment:
Why do you have else statement here? can we always have coco metric?

Reply:
For evaluating instance masks, `COCOEvaluator` expects an annotation_file since the eval dataloader does not return instance mask groundtruths. Also, the task config supports only a single annotation_file, so we run COCOEvaluator only when `training=False`

Reply:
Oh sorry, no keep it as is. I thought this is the else statement for "if self.task_config.segmentation_evaluation.report_train_mean_iou:"